src/cascadia/TerminalSettingsModel/FileUtils.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "FileUtils.h"
+
+#include <appmodel.h>
+#include <shlobj.h>
+#include <WtExeUtils.h>
+
+static constexpr std::string_view Utf8Bom{ u8"\uFEFF" };
+static constexpr std::wstring_view UnpackagedSettingsFolderName{ L"Microsoft\\Windows Terminal\\" };
+
+namespace Microsoft::Terminal::Settings::Model
+{
+    std::filesystem::path GetBaseSettingsPath()
+    {
+        static std::filesystem::path baseSettingsPath = []() {
+            wil::unique_cotaskmem_string localAppDataFolder;
+            // KF_FLAG_FORCE_APP_DATA_REDIRECTION, when engaged, causes SHGet... to return
+            // the new AppModel paths (Packages/xxx/RoamingState, etc.) for standard path requests.
+            // Using this flag allows us to avoid Windows.Storage.ApplicationData completely.
+            THROW_IF_FAILED(SHGetKnownFolderPath(FOLDERID_LocalAppData, KF_FLAG_FORCE_APP_DATA_REDIRECTION, nullptr, &localAppDataFolder));
+
+            std::filesystem::path parentDirectoryForSettingsFile{ localAppDataFolder.get() };
+
+            if (!IsPackaged())
+            {
+                parentDirectoryForSettingsFile /= UnpackagedSettingsFolderName;
+            }
+
+            // Create the directory if it doesn't exist
+            std::filesystem::create_directories(parentDirectoryForSettingsFile);
+
+            return parentDirectoryForSettingsFile;
+        }();
+        return baseSettingsPath;
+    }
+
+    std::string ReadUTF8File(const std::filesystem::path& path)

```

Comment:
This function unifies our many, many manual ways to read files from the disk.
This ensures we focus unsafe API usage in a single function and can easily introduce features like:
* Preventing us from returning cut-off files / files that were changed while reading
* Retrying in case something goes wrong temporarily.